apps/email/js/cards/lst/msg_click.js

```diff
@@ -0,0 +1,162 @@
+  // TODO: If UX designers want the size included in the warning
+  // message, add it here.
+  ConfirmDialog.show(dialog,
+    { // Confirm
+      id: 'msg-large-message-ok',
+      handler: function() { cb(true); }
+    },
+    { // Cancel
+      id: 'msg-large-message-cancel',
+      handler: function() { cb(false); }
+    }
+  );
+}
+
+/**
+ * A mixin for lst cards that want tapping on a message to go to a message
+ * reader. Check all the `this` references below to see implicit dependencies
+ * on instance state.
+ * A longer range TODO is to remove or make explicit the dependencies.
+ */
+  return {
+    onClickMessage: function(messageNode, event) {
+      // You cannot open a message if this is the outbox and it is syncing.
+      if (this.curFolder &&
+          this.curFolder.type === 'outbox' && this.outboxSyncInProgress) {
+        return;
+      }
+
+      var header = messageNode.message;
+
+      // Skip nodes that are default/placeholder ones.
+      if (header && header.isPlaceholderData) {
+        return;
+      }
+
+      if (this.editMode) {
+        var idx = this.selectedMessages.indexOf(header);
+        if (idx !== -1) {
+          this.selectedMessages.splice(idx, 1);
+        }
+        else {
+          this.selectedMessages.push(header);
+        }
+        this.setMessageChecked(messageNode, idx === -1);
+        this.selectedMessagesUpdated();
+        return;
+      }
+
+      if (this.curFolder && this.curFolder.type === 'localdrafts') {
+        var composer = header.editAsDraft(function() {
+          cards.pushCard('compose', 'animate', {
+            model: this.model,
+            composer: composer
+          });
+        });
+        return;
+      }
+
+      // When tapping a message in the outbox, don't open the message;
+      // instead, move it to localdrafts and edit the message as a
+      // draft.
+      if (this.curFolder && this.curFolder.type === 'outbox') {
+        // If the message is currently being sent, abort.
+        if (header.sendStatus.state === 'sending') {
+          return;
+        }
+        var draftsFolder =
+              this.model.foldersSlice.getFirstFolderWithType('localdrafts');
+
+        console.log('outbox: Moving message to localdrafts.');
+        this.model.api.moveMessages([header], draftsFolder, function(moveMap) {
+          header.id = moveMap[header.id];
+          console.log('outbox: Editing message in localdrafts.');
+          var composer = header.editAsDraft(() => {
+            cards.pushCard('compose', 'animate', {
+              model: this.model,
+              composer: composer
+            });
+          });
+        });
+
+        return;
+      }
+
+      var model = this.model,
+          headerCursor = this.headerCursor;
+
+      function pushMessageCard() {
+        cards.pushCard(
+          'message_reader', 'animate',
+          {
+            model: model,
+            headerCursor: headerCursor,
```

Comment:
available es2015 magic reminder / stylistic consensus question: we can just write "model, headerCursor, header" here (still on separate lines if desired). I've adopted this in GELAM, although I don't think I care.  I mainly just feel like a sucker for typing things twice when I could only type it once.

Having said that, it also simplifies things now that I'm starting to pass arguments through with destructuring.  Ex:
```js
function bob({ hat, cat, bat }) {
}
function mob() {
  let hat = 1, cat = 2, bat = 3;
  bob({ hat, cat, bat });
}
```

(And I've started to do that for the cases where the arg-list starts having more than 3 items or multiple booleans passed.)